src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2730,7 +2727,6 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
             UnaryOperatorKind resultOperatorKind = signature.Kind;
             var resultConstant = FoldUnaryOperator(node, resultOperatorKind, resultOperand, resultType, diagnostics);
 
-            CheckNativeIntegerFeatureAvailability(resultOperatorKind, node, diagnostics);
```

Comment:
>CheckNativeIntegerFeatureAvailability [](http://example.com/codeflow?start=12&length=37)

Should the checks be removed conditionally, i.e. only runtime flavor is used?

Reply:
Tagging @jaredpar for thought. My understanding is that we want to remove the check unconditionally.
The new compiler comes with a new BCL, so the runtime feature flag is present anyways.

Reply:
> The new compiler comes with a new BCL, so the runtime feature flag is present anyways.

I am not sure if the conclusion is accurate. It is still supported to target older runtimes, I think. In that case, behavior for the native integer feature will be "broken", as changes in NativeIntegerTests,cs indicate.

Reply:
It should be fairly easy to check if ```RuntimeSupportsNumericIntPtr``` is true. If so, then the check can be skipped.